### PR TITLE
Update DG: Add Use Cases

### LINKED ISSUE #30: Update DG: Add Use Cases

### PR DESCRIPTION
**Pull request by voldiemort**
Fixes #30

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -320,7 +320,24 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
 
-**Use case: Delete a person**
+**Use case: UC01 - Search for person(s)**
+
+**MSS**
+
+1.  User searches for a person by name
+2.  AddressBook shows a list of persons which fit the criteria
+
+    Use case ends.
```
Comment (seah-minlong) [inline docs/DeveloperGuide.md:330]: image

The formatting seems a little different from the examples given. Did they mention anything about it or is it ok

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -343,6 +362,51 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+
+**Extensions**
+
+* 1a. User could also search for persons by name/tag/role... etc. (UC01)
+
+* 2a. The list is empty.
+
+  Use case ends.
```
Comment (seah-minlong) [inline docs/DeveloperGuide.md:407]: image
Same formatting query about the extension

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (voldiemort) [reply docs/DeveloperGuide.md:330]: I followed the formatting present from the sample use case in the sample DG, so I think should be fine

Comment (seah-minlong) [reply docs/DeveloperGuide.md:330]: Okay sounds good 🥇

**Review by seah-minlong — APPROVED** [review]
LGTM

**Pull request merged**